docs/smart-contracts/creator-tools/events.mdx

```diff
@@ -0,0 +1,351 @@
+
+They also have the same sales settings across the contract unlike 1155 tokens.
+
+New tokens are created by calling the ZoraNFTCreatorV1 proxy contract.
+
+Once the contract is created, if the sale is active users can purchase tokens.
+
+We also support a `multicall` pattern with the `setupCalls` argument where the factory is granted temporary admin permissions to execute multiple commands on the contract after deployment allowing for setting additional settings or minting upon deployment.
+
+#### Creating an Edition:
+
+```sol
+function createEditionWithReferral(
+    string memory name,
+    string memory symbol,
+    address defaultAdmin,
+    uint64 editionSize,
+    uint16 royaltyBPS,
+    address payable fundsRecipient,
+    bytes[] memory setupCalls,
+    IMetadataRenderer metadataRenderer,
+    bytes memory metadataInitializer,
+    address createReferral
+)
+```
+
+#### Creating a Drop:
+
+Event Emitted when a drop is created from the factory:
+
+```
+event CreatedDrop(address indexed creator, address indexed editionContractAddress, uint256 editionSize)
+```
+
+This is emitted by the `ZoraNFTCreatorV1` factory contract.
+
+##### Setting a Price
+
+Event emitted with Sales Configuration Setup:
+
+```
+event SalesConfigChanged(address indexed changedBy);
+```
+
+After this event is emitted, the contract sales information can be queried and stored. See [subgraph implementation](https://github.com/ourzora/zora-creator-subgraph/blob/HEAD/src/ERC721Mappings/templates/ERC721DropMappings.ts#L56).
+
+##### Collecting a Token
+
+First, sales information can be retrieved by calling `salesConfig()` on the 721 contract which returns all of the presale (allowlist), and public sale (standard purchase) configuration.
+
+After this call, the `function mintWithRewards(address recipient, uint256 quantity, string calldata comment, address mintReferral)` function can be called.
+
+The mint fee can be queried from `zoraFeeForAmount(uint256 amount) returns (address, uint256 fee)` which returns the total mint fee for a given amount.
+
+The value sent is `pricePerToken * numberOfTokens + mintFee`.
+
+This emits:
+
+```sol
+event IERC721Drop.Sale(
+    address recipient,
+    uint256 quantity,
+    uint256 pricePerToken,
+    uint256 firstPurchasedTokenId
+);
+```
+
+and if salesComment is not unset (in solidity, is not an empty string) `''`:
+
+```
+event IERC721Drop.MintComment(
+    address sender, // Address sending the mint
+    address tokenContract, // Current NFT contract
+    uint256 tokenId,
+    uint256 quantity,
+    string comment
+);
+```
+
+## Protocol Rewards
+
+Zora protocol rewards power rewards for both the 721 and 1155 contracts.
+
+The contracts include a shared solidity library to calculate the fees and call a shared hyperstructure to escrow and store the deposits for users to deposit.
+
+This allows us to have the gas savings and security of a pull contract while also making it easy for users to retrieve all their rewards in one place.
```

Comment:
did you mean `full` instead of `pull` here?

Reply:
I explained this a little better to refer to `pull` vs `push` payments.